app/src/main/java/org/dicekeys/app/RecipeBuilder.kt

```diff
@@ -13,9 +14,21 @@ class RecipeBuilder constructor(val type: DerivationOptions.Type, val template:
         private set
     var purpose: String? = null
         private set
-    private var sequence = 1
-    private var lengthInChars = 0
-    private var lengthInBytes = 0
+    var rawJson: String? = null
+        private set
+
+    var sequence = 1
+        private set
+    var lengthInChars = template?.lengthInChars ?: 0
+        private set
+    var lengthInBytes = template?.lengthInBytes ?: 0
```

Comment:
Why is default 0?

Reply:
It's not an optional variable and needs initialization, default value 0 means unused in DerivationRecipe constructor.